integration_tests/src/main/python/string_test.py

```diff
@@ -836,6 +836,20 @@ def test_regexp_extract_idx_0():
                 'regexp_extract(a, "^([a-d]*)[0-9]*([a-d]*)\\z", 0)'),
         conf=_regexp_conf)
 
+def test_word_boundaries():
+    gen = mk_str_gen('([abcd]{1,3}[\n\t\r\f ]{0,2}[efgh]){1,5}')
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark: unary_op_df(spark, gen).selectExpr(
+                'rlike(a, "\\b")',
```

Comment:
I am a little confused by this. I manually tested rlike, regexp_extract and regexp_replace with strings that match the string gen pattern given and none of them match. rlike is always false, regexp_extract always returns an empty string and regexp_replace does not change the input string at all.  This is the same on both the CPU and the GPU.

My main concern is that we are not really testing these because we don't have anything where they match. Perhaps these patterns just never match on Spark in these situations. If so, then we should document it. If not then I really would like a test that causes them to match.

Reply:
Actually ti turned out to be user error on my part in scala. I needed to double escape the \b and \B, i.e. \\B. Now I really just want to make sure that you have verified that it is hitting the code path you want. :)

Reply:
Yes I had to change all of these to be double escaped to make the patterns work.

```
>>> spark.createDataFrame([("a e", "b a")], schema='a string,b string').selectExpr('regexp_replace(a, "\\\\b", "*") as works', 'regexp_replace(a, "\\b", "*") as not').show()

+-------+---+
|  works|not|
+-------+---+
|*a* *e*|a e|
+-------+---+
```

Also replacing boundaries which are not a character with an empty string is a NOOP. You really should replace them with something like '*' or anything that would make them appear.

Reply:
In that case, it looks like we need to change all character classes/string anchors supported by cuDF to use the double escape `\\\\` for every integration test here, correct?

Reply:
I think so, but I am not 100% sure. I was running in the python repl so there could be some odd differences there. I would suggest that everywhere you suspect there may be some issues that you insert in a `data_gen.debug_df` to check that the output is exactly what you would expect and then if what I found is true, update the tests accordingly.

To keep the scope limited I would just update test new tests added for this PR and then you can file a follow on BUG and/or PR to fix the other tests.

Reply:
I agree with changing all the integration tests. In my testing of other issues, I have confirmed the necessity to double escape any regular expression specific escape character in order to do the test properly (I *think* this is a Python specific behavior). Let's handle this in a follow-on issue.

Reply:
@anthony-chang can you change at least this integration test to double-escape the `b` and `B` and add the `*` (or any other character replacement) that Bobby suggested?